Uzira.rte/Actors/Brains/Uzira/Uzira.lua

```diff
@@ -0,0 +1,80 @@
+function Create(self)
+	self.minion = CreateAHuman("Skeleton", "Uzira.rte");
+	self.weapons = {"Uzira.rte/Blunderpop", "Uzira.rte/Blunderbuss", "Uzira.rte/Musket"};
+	self.spawnRadius = math.max(FrameMan.PlayerScreenWidth, FrameMan.PlayerScreenHeight) * 0.1;
+	self.boundsRadius = self.spawnRadius * 4 + self.minion.Radius;
+	self.spawnTerrainTolerance = 70;
+	self.spawnCost = self.minion:GetGoldValue(0, 1, 1);
+	
+	self.baseSpawnTime = 1000;
+	self.spawnTime = 1000;
+	self.spawnTimer = Timer();
+	self.minion.Team = self.Team;
+	self.minion.PinStrength = self.minion.Mass;
+	self.minion:AddScript("Uzira.rte/Actors/Shared/Undead.lua");
+	self.minions = {};
+end
+function Update(self)
+	if self.spawnTimer:IsPastSimMS(self.spawnTime) then
+		self.spawnTimer:Reset();
+		self.spawnTime = self.baseSpawnTime;
+		local funds = ActivityMan:GetActivity():GetTeamFunds(self.Team);
+		for i = 1, #self.minions do
+			local minion = self.minions[i];
+			if not MovableMan:IsActor(minion) then
+				table.remove(self.minions, i);
+			elseif minion.Health > 0 and SceneMan:ShortestDistance(self.Pos, minion.Pos, SceneMan.SceneWrapsX).Magnitude > self.boundsRadius then
+				minion.Health = 0;
+				for attachable in minion.Attachables do
+					if math.random() < 0.5 then
+						minion:RemoveAttachable(attachable, true, true);
+					end
+				end
+				--ActivityMan:GetActivity():ReportDeath(self.Team, -1);	--Disregard deaths of spawned minions?
+			end
+		end
+		if funds > self.spawnCost then
+			local spawnPos = self.Pos + Vector(0, self.Radius + RangeRand(0, self.spawnRadius)):RadRotate(RangeRand(-1, 1));
+			if SceneMan:CastMaxStrengthRay(self.Pos, spawnPos, 8) < self.spawnTerrainTolerance * 2 then
+
+				local pointCount = 10;
+				local radius = self.minion.Radius/math.sqrt(pointCount);
+				local totalSurroundingTerrain = 0;
+				local totalTolerance = self.spawnTerrainTolerance * pointCount;
+				
+				for i = 0, pointCount do
+					local checkPos = spawnPos + Vector(math.sqrt(i) * radius, 0):RadRotate(i * 2.39996);
+					local terrCheck = SceneMan:GetTerrMatter(checkPos.X, checkPos.Y);
+					if terrCheck == rte.airID then
+						totalSurroundingTerrain = totalSurroundingTerrain + self.spawnTerrainTolerance * 2;	--Treat air as unsuitable terrain to spawn from
+					else
+						totalSurroundingTerrain = totalSurroundingTerrain + SceneMan:GetMaterialFromID(terrCheck).StructuralIntegrity;
+					end
+					if totalSurroundingTerrain > totalTolerance then
+						break;
+					end
+				end
+				if totalSurroundingTerrain < totalTolerance then
+					local weapon = CreateHDFirearm(self.weapons[math.random(#self.weapons)]);
+					weapon.DeleteWhenRemovedFromParent = true;
+					local skeleton = self.minion:Clone();
+					skeleton:AddInventoryItem(weapon);
+					skeleton.Pos = spawnPos;
+					skeleton.HFlipped = self.HFlipped;
+					skeleton.RotAngle = RangeRand(-1.5, 1.5);
+					skeleton.AIMode = math.random(Actor.AIMODE_SENTRY, Actor.AIMODE_GOTO);
```

Comment:
Necromancy works great now! That said, if you're up for it, I think it might be nice to do a bit more complex stuff here (feel free to discuss in dms)

Uzira could have pie buttons that let her control the skeletons to a certain degree (a sub-pie menu would probably be good for this + turning on/off necromancy).
Skeletons would default to spawning in sentry or patrol mode. She can toggle this to have them spawn following her instead. She can also use a button to send all current skeletons to charge the closest enemy (or maybe to brainhunt, or maybe that should be a different button, I'm not sure.
I guess skeletons that are in attack mode wouldn't despawn if they're too far from uzira. Maybe instead they could get a buff, but die over time, so skeleton assaults last for a limited time.

With the ability to give general orders to these skeletons, they could
1. Not have a HUD again
2. Not be player controllable, since you can actually do things with them now, and that'll heavily differentiate them from purchased troops

Reply:
I think as long as they are running the Lua AI, disbanding a squad should set them to the leader's AI mode. The feature is there for all actors, it's just not very apparent. 

Skeletons have a 33% chance of spawning in GOTO which either sends them off at a nearby enemy or sets them to follow Uzira. I should allow a non-player Uzira to send off enemies every now and then though. What I could settle with is probably something like incrementally diminishing the skellies' health as they get further from the brain. 

1. The HUD thing - I played a short bit of Conquest against Uzira and it feels dissonant to have units both with visible and unvisible HUD roaming around. I'd say it has to be either or. 
2. I have no idea if this is even possible without cpp support?

Reply:
1. I guess we could not use skeletons in loadouts (maybe even make them unbuyable and only spawn from necromancy), or have some slightly different looking skeletons for this, or give them some other visual effect to differentiate them. I think, though, that if 2. is done, it'll be less weird?

2. is definitely doable, I did it years ago, though I don't remember the details haha. I can look into it.

Anyway, in general you get what I'm saying? Let uzira not control these skelies directly but instead give them broad-scale commands, I think it'd be neat to make Uzira faction seem even more different and special!

Reply:
I don't know how much of a UI workload it would be so I think I'll refrain from it for now, feel free to take a shot at it if you like :p

I also think the zombies are ugly, but I guess it would make more sense for *them* to come by craft, from somewhere where there is a clone vat? I don't know